WordPress/gutenberg
# Components: Deprecate `COLORS.white`

Part of #44116, #43997

## What?

Deprecates the internal `COLORS.white` variable, replacing most of the usages to theme-ready variables.

There are a few more usages in the package, but they are each a bit special so we'll deal with them separately.

## Why?

`COLORS.white` is not theme-ready, and we should prevent new usages in component styles.


## Testing Instructions

Check in Storybook that there are no visual regressions in the following component states. Use the theme tool in the toolbar to toggle to an example dark mode.

<img src="https://github.com/user-attachments/assets/da701756-8471-44c1-92ae-7136f7fc4e55" alt="Theme tool in toolbar" width="334">

- Menu — Active item text color
- Navigation — Active item text color. There's a style bug on hover of the active item, but not fixing here since it's deprecated and an pre-existing issue.
- ToggleGroupControl — Selecting different items. Also check with icon items and in `isDeselectable` mode.
